### DISCUSSION TITLE
Trying to learn how to use this

### DISCUSSION THREAD

**Discussion by Steven121** [nginx-proxy/nginx-proxy#1944] [Q&A]
I have a Linux server on AWS.  I'm just trying to set up nginx via Docker on Ubuntu as a test.

Is it ok to have nginx not be running on the Linux server outside of the Docker container?

Do I need to have DNS set up and use the VIRTUAL_HOST?  If so, would modifying the /etc/hosts file be sufficient?  Or do I need to update the named.conf file or something else?  

This is what we are supposed to do to run nginx via Docker:

$ docker run -d -p 80:80 -v /var/run/docker.sock:/tmp/docker.sock:ro jwilder/nginx-proxy

My docker.sock file is in /run/

What should jwilder/nginx-proxy be changed to?  Should anything else be changed?  Port 80 is fine to run inside the Docker instance and outside (on the Linux OS).

**Comment by Stellaverse** [comment]
Hi Steven,

It's much easier than you're imagining. Say you have a domain like www.myapp.com. You would setup the DNS for www.myapp.com to point to your server IP (or to you ELB, which forwards the requests to you Autoscaling-Group).

Once that's setup then you're pretty much good to go. You would just do something like this:

```
docker run -d -p 80:80 -v /var/run/docker.sock:/tmp/docker.sock:ro jwilder/nginx-proxy

docker run -d -e 'VIRTUAL_HOST=www.myapp.com' yourdockerhubname/www-myapp
```

And you should be good to go.

You can also run multiples of the same container behind `nginx-proxy` by declaring the same `VIRTUAL_HOST` var and `nginx-proxy` will pick up on the new containers automagically.

**Comment by Stellaverse** [comment]
And yes, it's fine to not have `Nginx` running on the host instance. That's not necessary. Nginx is fully encapsulated in the `nginx-proxy` container.

**Comment by vfulco** [comment]
Trying to understand this statement better..."You can also run multiples of the same container behind nginx-proxy by declaring the same VIRTUAL_HOST var and nginx-proxy will pick up on the new containers automagically."

How would nginx-proxy then differentiate between containers then?  I also notice your line to launch nginx-proxy didn't contain a default_host?

**Comment by Stellaverse** [comment]
You just start a bunch of contenders with the same `VIRTUAL_HOST=www.myapp.com` and then the `Dockergen` component of `nginx-proxy` will detect the IP of any given container and add its IP to the nginx conf. I believe it then uses simple round-robin load balancing to send requests to any IP listed under any given domain.

**Comment by jsdevel** [comment]
I have a question...

How can I have `nginx-proxy` reverse proxy a service running on the host machine?  My scenario is this:
- I'm setting up a CI server on an internal network that will host n number of sites.
- I have jenkins installed on my host machine (not in a container) because I'll need to build docker images.  I saw some articles about DIND, but it seems overly complicated / not directly supported.
- Jenkins is running on 127.0.0.1:8080 and I'd like it to be accessible externally via `jenkins.internal.com` on port 80.

Is there a way to add a vhost to `nginx-proxy` to accomplish this?

**Comment by wader** [comment]
@jsdevel i would try to add a static `server {...}` config file to `/etc/nginx/conf.d` (see "Proxy-wide" in README) which proxies to `127.0.0.1:8080`
